### PR TITLE
[Php80] Handle FQCN @\ usage on AnnotationToAttributeRector

### LINKED ISSUE #8301: Incorrect behavior of AnnotationToAttributeRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/d2786e96-276d-420b-892e-55cc2078cc1e

```php
<?php

declare(strict_types = 1);

namespace App\Model\User;

/**
 * @\Doctrine\ORM\Mapping\Entity()
 * @\Doctrine\ORM\Mapping\Table("user", indexes={
 *  @\Doctrine\ORM\Mapping\Index(name="name_index", columns={"name"}),
 *  @\Doctrine\ORM\Mapping\Index(name="surname_index", columns={"surname"}),
 * })
 * @\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity("azureB2cUuid")
 * @\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity("uuid")
 * @\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity("email")
 */
class Entity {
    
    /**
     * @\Symfony\Component\Validator\Constraints\NotBlank()
     * @\Symfony\Component\Validator\Constraints\Email(mode="strict")
     * @\Doctrine\ORM\Mapping\Column(type="string", unique=true)
     * @\Symfony\Component\Serializer\Annotation\Groups({"import", "export-user", "export-claim"})
     */
    protected string $email = "";
}
```

### Responsible rules

* `AnnotationToAttributeRector`

## Expected Behavior

Annotations shouldn't be removed but converted to attributes

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8301

### CODE REVIEW

File: tests/Issues/FqcnAnnotationToAttribute/Fixture/fixture.php.inc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('azureB2cUuid')]
+#[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('uuid')]
+#[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('email')]
+class Entity {
+{
+    #[\Symfony\Component\Validator\Constraints\NotBlank]
+    #[\Symfony\Component\Validator\Constraints\Email(mode: 'strict')]
+    #[\Doctrine\ORM\Mapping\Column(type: 'string', unique: true)]
+    #[\Symfony\Component\Serializer\Annotation\Groups(['import', 'export-user', 'export-claim'])]
+    protected string $email = "";
```
Comment (samsonasik) [inline tests/Issues/FqcnAnnotationToAttribute/Fixture/fixture.php.inc]: @ZZromanZZ this the failing fixture for issue:

- https://github.com/rectorphp/rector/issues/8301

Could you verify this is the expected output?

Comment (ZZromanZZ) [reply tests/Issues/FqcnAnnotationToAttribute/Fixture/fixture.php.inc]: Yes, this is expected

Comment (samsonasik) [reply tests/Issues/FqcnAnnotationToAttribute/Fixture/fixture.php.inc]: thank you for verify, I will look into it

**Comment by samsonasik** [comment]
Progress 👍 , after rebase of PR:

- https://github.com/rectorphp/rector-src/pull/5271

it now only `@Table` that not converted:

```diff
There was 1 failure:

1) Rector\Core\Tests\Issues\FqcnAnnotationToAttribute\FqcnAnnotationToAttributeTest::test with data set #0
Failed on fixture file "fixture.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 namespace Rector\Core\Tests\Issues\FqcnAnnotationToAttribute\Fixture;
 
 #[\Doctrine\ORM\Mapping\Entity]
-#[\Doctrine\ORM\Mapping\Table('user')]
 #[\Doctrine\ORM\Mapping\Index(name: 'name_index', columns: ['name'])]
 #[\Doctrine\ORM\Mapping\Index(name: 'surname_index', columns: ['surname'])]
 #[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('azureB2cUuid')]
```

that seems due to `NestedAnnotationToAttribute` usage

**Comment by samsonasik** [comment]
@ZZromanZZ Fixed 🎉

**Comment by ZZromanZZ** [comment]
Wow, thank you :+1:

File: tests/Issues/FqcnAnnotationToAttribute/Fixture/fixture.php.inc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Rector\Core\Tests\Issues\FqcnAnnotationToAttribute\Fixture;
+
+#[\Doctrine\ORM\Mapping\Entity]
+#[\Doctrine\ORM\Mapping\Index(name: 'name_index', columns: ['name'])]
+#[\Doctrine\ORM\Mapping\Index(name: 'surname_index', columns: ['surname'])]
+#[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('azureB2cUuid')]
+#[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('uuid')]
+#[\Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity('email')]
+#[\Doctrine\ORM\Mapping\Table('user')]
```
Comment (samsonasik) [inline tests/Issues/FqcnAnnotationToAttribute/Fixture/fixture.php.inc:38]: this `#[\Doctrine\ORM\Mapping\Table('user')]` on the last attribute seems due to doctrine configuration structure, I think it doesn't matter :)

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba let's merge it ;)

**Pull request merged**